src/menu/create_wallet_menu.c

```diff
@@ -0,0 +1,145 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "create_wallet_menu.h"
+
+#include "constant_texts.h"
+#include "ui_menu.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+typedef enum {
+  GENERATE_NEW_WALLET,
+  RESTORE_FROM_SEED,
+} create_wallet_options_e;
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static const evt_config_t create_wallet_evt_config = {
+    .abort_disabled = false,
+    .evt_selection.byte = EVT_CONFIG_UI_EVT,
+    .timeout = MAX_INACTIVITY_TIMEOUT};
+
+const flow_step_t create_wallet_step = {
+    .step_init_cb = create_wallet_menu_initialize,
+    .p0_cb = NULL,
+    .ui_cb = create_wallet_menu_handler,
+    .usb_cb = NULL,
+    .nfc_cb = NULL,
+    .evt_cfg_ptr = &create_wallet_evt_config,
```

Comment:
either make the definition for `evt_config_t create_wallet_evt_config` common for all the menu navigation steps as they stay consistent everywhere. Otherwise, `evt_cfg_ptr`  can be converted to instance instead of pointer.